### ISSUE TITLE
SelectCheckboxMenu: First selectItem is falsely selected when opening panel with space bar

### ISSUE THREAD

**Issue by miedrl** [primefaces/primefaces#8340]
**Describe the defect**
When navigating the site with the keyboard instead of the mouse, SelectCheckboxMenus can be opened and closed with space bar. When clicking space bar, the component opens to show the selectItems but the first selectItem is instantly selected as it also reacts to space bar.

**Environment:**
 - PF Version: _10.0.0_

**To Reproduce**
1. Add the XHTML code below to your example site.
2. Use tab key until the SelectCheckboxMenu is focused, then press space bar to open the menu.
3. See error > by pressing space bar, the first selectItem is selected

**Expected behavior**
When opening a selectCheckboxMenu with the space bar, the focused selectItem should not be selected instantly, only after pressing space bar a second time.

**Example XHTML**
```html
<h:form id="frmTest">
   <p:selectCheckboxMenu id="test"
      styleClass="p-col-12"
      value="#{ShowcaseView.viewData.selectManyMenuArr[0]}"
      filter="false">
              
      <f:selectItem itemLabel="Option1" itemValue="Option1" />
      <f:selectItem itemLabel="Option2" itemValue="Option2" />
      <f:selectItem itemLabel="Option3" itemValue="Option3" />
      <f:selectItem itemLabel="Option4" itemValue="Option4" />
   </p:selectCheckboxMenu>
</h:form>
```

**Comment by melloware** [comment]
@miedrl can you try this MonkeyPatch and see if your issue is fixed?  If so I will submit a PR.

**MonkeyPatch for PF11:**

```javascript
if (PrimeFaces.widget.SelectCheckboxMenu) {
    PrimeFaces.widget.SelectCheckboxMenu.prototype.bindKeyEvents = function() {
        var $this = this;

        this.keyboardTarget.on('focus.selectCheckboxMenu', function() {
            $this.jq.addClass('ui-state-focus');
            $this.menuIcon.addClass('ui-state-focus');
        }).on('blur.selectCheckboxMenu', function() {
            $this.jq.removeClass('ui-state-focus');
            $this.menuIcon.removeClass('ui-state-focus');
        }).on('keydown.selectCheckboxMenu', function(e) {
            var keyCode = $.ui.keyCode,
                key = e.which;

            if (!$this.isLoaded()) {
                $this._renderPanel();
            }

            switch (key) {
                case keyCode.ENTER:
                case keyCode.SPACE:
                    if ($this.panel.is(":hidden")) {
                        $this.show();
                        e.stopPropagation(); // GitHub #8340
                    } else
                        $this.hide();

                    e.preventDefault();
                    break;

                case keyCode.DOWN:
                    if (e.altKey) {
                        if ($this.panel.is(":hidden"))
                            $this.show();
                        else
                            $this.hide();
                    }

                    e.preventDefault();
                    break;

                case keyCode.TAB:
                    if ($this.panel.is(':visible')) {
                        if (!$this.cfg.showHeader) {
                            $this.itemContainer.children('li:not(.ui-state-disabled):first').find('div.ui-helper-hidden-accessible > input').trigger('focus');
                        } else {
                            $this.toggler.find('> div.ui-helper-hidden-accessible > input').trigger('focus');
                        }
                        e.preventDefault();
                    }

                    break;

                case keyCode.ESCAPE:
                    $this.hide();
                    break;
            };
        });
    }
};
```

**Comment by miedrl** [comment]
@melloware I tried your patch. It throws a "$this.isLoaded is not a function" error but that could be because I'm missing something in my code. Despite the error, the patch fixes my problem just fine.

**Comment by melloware** [comment]
My patch above was for PF 11. I will try and make one for pf10

**Comment by melloware** [comment]
@miedrl PrimeFaces 10 MonkeyPatch:

```javascript
if (PrimeFaces.widget.SelectCheckboxMenu) {
    PrimeFaces.widget.SelectCheckboxMenu.prototype.bindKeyEvents = function() {
        var $this = this;

        this.keyboardTarget.on('focus.selectCheckboxMenu', function() {
            $this.jq.addClass('ui-state-focus');
            $this.menuIcon.addClass('ui-state-focus');
        }).on('blur.selectCheckboxMenu', function() {
            $this.jq.removeClass('ui-state-focus');
            $this.menuIcon.removeClass('ui-state-focus');
        }).on('keydown.selectCheckboxMenu', function(e) {
            var keyCode = $.ui.keyCode,
                key = e.which;

            if ($this.cfg.dynamic && !$this.isDynamicLoaded) {
                $this._renderPanel();
            }

            switch (key) {
                case keyCode.ENTER:
                case keyCode.SPACE:
                    if ($this.panel.is(":hidden")) {
                        $this.show();
                        e.stopPropagation(); // GitHub #8340
                    } else
                        $this.hide();

                    e.preventDefault();
                    break;

                case keyCode.DOWN:
                    if (e.altKey) {
                        if ($this.panel.is(":hidden"))
                            $this.show();
                        else
                            $this.hide();
                    }

                    e.preventDefault();
                    break;

                case keyCode.TAB:
                    if ($this.panel.is(':visible')) {
                        if (!$this.cfg.showHeader) {
                            $this.itemContainer.children('li:not(.ui-state-disabled):first').find('div.ui-helper-hidden-accessible > input').trigger('focus');
                        } else {
                            $this.toggler.find('> div.ui-helper-hidden-accessible > input').trigger('focus');
                        }
                        e.preventDefault();
                    }

                    break;

                case keyCode.ESCAPE:
                    $this.hide();
                    break;
            };
        });
    }
};
```
